src/main/java/com/google/sps/algorithms/MatchRanking.java

```diff
@@ -63,35 +73,41 @@ public int compare(Object o1, Object o2) {
 
     /**
     * Function that takes a bio and maps of the word counts in saved bios and all bios, and returns the score for the 
-    * bio as a double.
+    * bio as a double. The score is the nth root of the regular score to normalize larger bios.
     */
     private static double calculateBioScore(String bio, Map<String, Integer> savedMatchesWordCount, Map<String, Integer> allUserWordCount) {
         //separate bio words by whitespace and store in an array
         String[] bioWords = bio.toLowerCase().split("\\W+");
 
         double numerator = calculateBioProbability(bioWords, savedMatchesWordCount);
         double denominator = calculateBioProbability(bioWords, allUserWordCount);
+        double result = numerator / denominator;
+
+        //if result == NaN, return 0
+        if(result != result) {
+            return 0.0;
+        }
 
-        return (numerator/denominator);
+        double nthRoot = Math.pow(result, (1.0/bioWords.length));
```

Comment:
nit: Add a comment explaining that this is meant to normalize it so there's no bias towards longer/shorter bios?

Reply:
Done!